src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -380,7 +429,8 @@ public void cleanup(Long id, boolean deleteRepositories) {
         try {
             // create a large zip file with all zipped repos and provide it for download
             log.debug("Create zip file for all repositories");
-            zipFilePath = Paths.get(zippedRepoFiles.get(0).getParent().toString(), exercise.getCourse().getTitle() + " " + exercise.getTitle() + studentIds.hashCode() + ".zip");
+            zipFilePath = Paths.get(zippedRepoFiles.get(0).getParent().toString(),
+                    exercise.getCourse().getTitle() + " " + exercise.getTitle() + (studentIds != null ? studentIds.hashCode() : "ALL") + ".zip");
```

Comment:
I wonder if we have to make this change, I would just use e.g. the timestamp as a hash here.

Reply:
In the current setup, we have to make this change, as the studentIds list can be null (therefore calling .hashCode() can produce a NullPointerException), but I agree that using a timestamp is better suited here.